cardano-api/src/Cardano/Api/MetaData.hs

```diff
@@ -123,49 +130,41 @@ jsonToMetadataValue  av =
           Right n -> Right (Api.TxMetaNumber n)
-      Aeson.String txt -> jsonToMetadataString txt
+      Aeson.String txt ->
+        case Text.stripPrefix bytesPrefix txt of
+          Nothing  -> jsonToMetadataString txt
+          Just hex -> case Base16.decode (Text.encodeUtf8 hex) of
+            (bytes, "") -> jsonToMetadataBytes bytes
+            (_, err)    -> Left $ ConversionErrBadBytes err
       Aeson.Array vs ->
-        let xs = Vector.toList vs in
-        case convertAsMap [] xs of
-          Nothing -> Api.TxMetaList <$> traverse jsonToMetadataValue xs
-          Just ys -> Right (Api.TxMetaMap ys)
+        Api.TxMetaList <$> traverse jsonToMetadataValue (Vector.toList vs)
       Aeson.Object kvs ->
-        if HashMap.keys kvs /= ["hex"]
-          then Api.TxMetaMap . List.sortOn fst <$> traverse convertPair (HashMap.toList kvs)
-          else
-            case HashMap.elems kvs of
-              [Aeson.String bs] ->
-                case Base16.decode (Text.encodeUtf8 bs) of
-                  (bs2, "") -> Right $ TxMetaBytes bs2
-                  (_, bsf) -> Left $ ConversionErrBadBytes bsf
-              _otherwise -> Left $ ConversionErrExpected "[Aeson.String x]" (show kvs)
+        Api.TxMetaMap . List.sortOn fst <$> traverse convertPair (HashMap.toList kvs)
   where
     convertPair :: (Text, Aeson.Value)
                 -> Either MetaDataJsonConversionError (TxMetadataValue, TxMetadataValue)
     convertPair (k, v) =
       (,) <$> jsonToMetadataValue (Aeson.String k) <*> jsonToMetadataValue v
 
-    -- If the list of values is a list of pairs, then return it as such, otherwise
-    -- return Nothing.
-    convertAsMap :: [(TxMetadataValue, TxMetadataValue)] -> [Aeson.Value]
-                    -> Maybe [(TxMetadataValue, TxMetadataValue)]
-    convertAsMap accum xs =
-      case xs of
-        [] -> Just $ List.sortOn fst accum
-        (Aeson.Array ys:zs) ->
-          case Vector.toList ys of
-            [a, b] -> case (jsonToMetadataValue a, jsonToMetadataValue b) of
-                        (Right am, Right bm) -> convertAsMap ((am, bm):accum) zs
-                        _otherwise -> Nothing
-            _otherwise -> Nothing
-        _otherwise -> Nothing
-
-
 jsonToMetadataString :: Text -> Either MetaDataJsonConversionError TxMetadataValue
 jsonToMetadataString txt =
-    if BS.length utf8 > 64
+    if BS.length utf8 > maxStringSize
       then Left ConversionErrLongerThan64Bytes
       else Right (Api.TxMetaText txt)
   where
     utf8 :: ByteString
     utf8 = Text.encodeUtf8 txt
+
+jsonToMetadataBytes :: ByteString -> Either MetaDataJsonConversionError TxMetadataValue
+jsonToMetadataBytes bytes =
+    if BS.length bytes > maxStringSize
+      then Left ConversionErrLongerThan64Bytes
+      else Right (Api.TxMetaBytes bytes)
+
+-- | JSON strings that are base16 encoded and prefixed with 'bytesPrefix' will
+-- be encoded as CBOR bytestrings.
+bytesPrefix :: Text
+bytesPrefix = "0x"
```

Comment:
This is a bad idea.

Reply:
May I ask why "it's a bad idea". 

Also, when "requesting changes" on a PR, it's better to tell the author what changes are expected.

Reply:
We used the `0x` and it made it difficult to distingush `ByteString` from `Text`. Property testing immediately found all the holes and the only way to fix it was to represent `ByteString`s an object like `{ "bytes" : "ffffff" }`.

Reply:
:sunglasses: property testing demonstrating things working fine.